src/Model/Behavior/LimitResultsBehavior.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Tatoeba Project, free collaborative creation of languages corpuses project
+ * Copyright (C) 2020 Tatoeba Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+namespace App\Model\Behavior;
+
+use Cake\ORM\Behavior;
+use Cake\ORM\Query;
+
+class LimitResultsBehavior extends Behavior
+{
+    private function getNeededAssociations(Query $query) {
+        $fields = [];
+        foreach (['where'] as $clause) {
+            $clause = $query->clause($clause);
```

Comment:
Is there a reason why you keep the `foreach` loop instead of getting the clause directly (i.e. `$clause = $query->clause('where');`)